community/index/src/main/java/org/neo4j/index/internal/gbptree/GBPTree.java

```diff
@@ -757,9 +765,12 @@ private void close( Header.Writer headerWriter ) throws IOException
 
         try
         {
-            // Perform a checkpoint before closing. If no changes has happened since last checkpoint,
-            // no new checkpoint will be created.
-            checkpoint( IOLimiter.unlimited(), headerWriter );
+            if ( health.isHealthy() )
```

Comment:
@tinwelint can you accept current way of logging, or you would prefer monitor instead? I'm fine with any of those, sure monitor have much wider range of applicability but since here we just need to log i think using just log is fine (and we have assertable log to be able to test that)